docker_cli/_docker_cuda.py

```diff
@@ -0,0 +1,284 @@
+                ENV CUDACXX=/usr/local/cuda-{version_name}/bin/nvcc
+
+                USER $DEFAULT_USER
+                """
+            ).strip()
+        )
+        return body
+
+    @abstractmethod
+    def _generate_initial_lines(self, repo_ver: str, *, arch: str = "x86_64") -> str:
+        """
+        Generates the first lines of the cuda install dockerfile.
+
+        These lines typically acquire the necessary CUDA repo and configure the
+        image to install CUDA.
+
+        Returns
+        -------
+        str
+            The initial lines of the CUDA install dockerfile.
+        """
+        ...
+
+    @abstractmethod
+    def generate_install_lines(self, build_targets: Iterable[str]) -> str:
+        """
+        Generates the install lines for the cuda install dockerfile.
+
+        These lines typically use the packet manager to install the given CUDA
+        libraries and then clean the packet manager cache.
+
+        Returns
+        -------
+        str
+            The install lines of the CUDA install dockerfile.
+        """
+        ...
+
+    @property
+    def package_manager(self) -> PackageManager:
+        """
+        The package manager used by this dockerfile generator.
+
+        This ensures that all subclasses of this class will have a package manager
+        property, but doesn't require the abstract base class to specify it.
+
+        Raises
+        ------
+        AttributeError
+            If no package manager is present on the object.
+        """
+        if not hasattr(self, "_package_manager"):
+            raise AttributeError(  # pragma: no cover
+                "Attribute 'package_manager' not found."
+            )
+        return getattr(self, "_package_manager")
+
+
+class AptGetCUDADockerfileGen(CUDADockerfileGenerator):
+    """
+    A CUDA dockerfile generator for debian-based apt-get systems
+    """
+
+    _dev_build_targets = [
+        "cuda-cudart-dev-$CUDA_PKG_VERSION",
+        "cuda-nvcc-$CUDA_PKG_VERSION",
+        "libcufft-dev-$CUDA_PKG_VERSION",
+    ]
+
+    def __init__(self, url_reader: URLReader):
+        super(AptGetCUDADockerfileGen, self).__init__(url_reader=url_reader)
+        self._package_manager: PackageManager = get_package_manager("apt-get")
+
+    def _generate_initial_lines(self, repo_ver: str, *, arch: str = "x86_64") -> str:
+        cuda_repo_name = (
+            "https://developer.download.nvidia.com/"
+            f"compute/cuda/repos/{repo_ver}/{arch}/"
+        )
+        filename = f"cuda-keyring_1.0-1_all.{self.package_manager.file_type}"
+        read_section = self.url_reader.generate_read_command(
+            target=f"{cuda_repo_name}" + filename,
+            output_file=filename,
+        )
+        package_command_section = self.package_manager.generate_package_command(
+            target=filename,
+        )
+        return f"RUN {read_section} \\\n && {package_command_section}"
```

Comment:
I think it makes sense to split this into two separate `RUN` commands.

In general, it's usually a good idea to break up our Dockerfile contents into lots of small, independent steps in order to make better use of the cache.

A notable exception to this is when one the commands is cleaning up from a previous step (like clearing the package manager's cache after installing a package) -- in that case, it's better to combine those two commands into a single `RUN` step in order to avoid bloating the docker image.

Another exception that's commonly recommended is to combine `apt-get update` and `apt-get install` into a single `RUN` step as a [cache-busting](https://docs.docker.com/develop/develop-images/dockerfile_best-practices/#apt-get) strategy.

In this case, there doesn't seem to be any reason to want to combine these two commands into a single `RUN` step, so I'd suggest separating them in to two separate steps:

```suggestion
        return textwrap.dedent(f"""
           RUN {read_section}
           RUN {package_command_section}
       """).strip()
```